src/MiddlewareInterface.php

```diff
@@ -2,15 +2,19 @@
 
 namespace Interop\Http\Server;
 
+use Interop\Http\Server\RequestHandlerInterface;
 use Psr\Http\Message\ResponseInterface;
 use Psr\Http\Message\ServerRequestInterface;
-use Psr\Http\Server\MiddlewareInterface as PsrInterface;
+use Psr\Http\Server\MiddlewareInterface as PsrMiddlewareInterface;
+use Psr\Http\Server\RequestHandlerInterface as PsrRequestHandlerInterface;
+
+class_alias(RequestHandlerInterface::class, PsrRequestHandlerInterface::class);
```

Comment:
This line will be executed only when `MiddlewareInterface` is autoloaded, does that work in every case? Maybe it would be safer to put this line in a file that is not autoloaded but rather always included by Composer? (`file` type in composer.json)

Reply:
yeh, it's not working always, you can't replace interface if it is already defined:

https://3v4l.org/9Kgdr

Reply:
This code does not fail:

```php
require 'vendor/autoload.php';

use Interop\Http\Server\RequestHandlerInterface;
use Interop\Http\Server\MiddlewareInterface;
use Psr\Http\Server\MiddlewareInterface as PsrMiddewareInterface;
use Psr\Http\Server\RequestHandlerInterface as PsrRequestHandlerInterface;
use Psr\Http\Message\ResponseInterface;
use Psr\Http\Message\ServerRequestInterface;

class_alias(RequestHandlerInterface::class, PsrRequestHandlerInterface::class);

class HandlerBar implements RequestHandlerInterface
{
    public function handle(ServerRequestInterface $request): ResponseInterface
    {
    }
}

class HandlerFoo implements PsrRequestHandlerInterface
{
    public function handle(ServerRequestInterface $request): ResponseInterface
    {
    }
}

class Bar implements MiddlewareInterface
{
    public function process(ServerRequestInterface $request, RequestHandlerInterface $handler): ResponseInterface
    {
        return $handler->handle($request);
    }
}

class Foo implements PsrMiddewareInterface
{
    public function process(ServerRequestInterface $request, RequestHandlerInterface $handler): ResponseInterface
    {
        return $handler->handle($request);
    }
}
```

So I guess placing the alias in a `file` to be sure is included always should work

Reply:
what do you have in your `composer.json` when you testing that code?

Reply:
```json
{
    "require": {
        "http-interop/http-server-middleware": "^1.1"
    }
}
```